File: Source Code/BusinessLogic/CarRental/CarRental/Model/CarList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace CarRental.Model
{
    public class CarList
    {
        public List<Car> cars;

        public CarList()
        {
            List<Car> carList = new List<Car>();
        }

        public void addCar(Car car)
        {
            cars.Add(car);
        }
        public void deleteCar(string regNumber)
        {
            for (int i = 0; i < cars.Capacity; i++)
            {
                if (cars[i].registrationNumber.Equals(regNumber))
                {
                    cars.RemoveAt(i);
                }
            }
        }
        public List<Car> returnAllCars(Car car)
        {
            return cars;
        }
    }
}

File: Source Code/AdministratorPresentationLayer/src/Model/CarList.java
package Model;

import java.util.ArrayList;

public class CarList {
    private ArrayList<Car> cars;

    public CarList() {
        cars =  new ArrayList<Car>();
    }

    public void addCar(Car car)
    {
        cars.add(car);
    }

    public void deleteCar(String registrationNumber)
    {
        for(int i=0; i<cars.size(); i++)
        {
            if(cars.get(i).getRegNumber().equals(registrationNumber))
            {
                cars.remove(i);
            }
        }
    }

    public void editCar(String registrationNumber,Car car)
    {
        for(int i=0; i<cars.size(); i++)
        {
            if(cars.get(i).getRegNumber().equals(registrationNumber))
            {
                cars.get(i).setRegNumber(car.getRegNumber());
                cars.get(i).setCarBrand(car.getCarBrand());
                cars.get(i).setCarModel(car.getCarModel());
                cars.get(i).setDateOfProduction(car.getYearOfProduction());
                cars.get(i).setGearBoxType(car.getGearBoxType());
                cars.get(i).setNumberOfDoors(car.getNumberOfDoors());
                cars.get(i).setEngineCapacity(car.getEngineCapacity());
                cars.get(i).setFuelType(car.getFuelType());
                cars.get(i).setCarColor(car.getCarColor());
                cars.get(i).setCarDescription(car.getCarDescription());
                cars.get(i).setCarPrice(car.getCarPrice());
                cars.get(i).setCarDeposit(car.getCarDeposit());

            }
        }
    }
    public int size()
    {
        return cars.size();
    }

}
